tests/python/gaia-ui-tests/gaiatest/apps/system/regions/time_picker.py

```diff
@@ -21,48 +17,57 @@
 class TimePicker(Base):
 
     _time_picker_locator = (By.CLASS_NAME, 'value-selector-time-picker')
-    _done_button_locator = (By.CSS_SELECTOR, '.value-selector-time-picker-buttons > button.value-selector-confirm')
-    _hour_picker_locator = (By.CLASS_NAME, 'value-picker-hours')
-    _minutes_picker_locator = (By.CLASS_NAME, 'value-picker-minutes')
-    _hour24_picker_locator = (By.CLASS_NAME, 'value-picker-hour24-state')
+    _done_button_locator = (By.CSS_SELECTOR,
+                            '.value-selector-time-picker-buttons > button.value-selector-confirm')
+
+    _hour_picker_locator = (By.CSS_SELECTOR, '.value-picker-hours > div')
+    _minutes_picker_locator = (By.CSS_SELECTOR, '.value-picker-minutes > div')
+    _hour24_picker_locator = (By.CSS_SELECTOR, '.value-picker-hour24-state > div')
 
     def __init__(self, marionette):
         Base.__init__(self, marionette)
         self.marionette.switch_to_frame()
-        Wait(self.marionette).until(expected.element_displayed(*self._hour_picker_locator))
+        self.wait_for_element_displayed(*self._current_element(*self._hour_picker_locator))
 
```

Comment:
We're supposed to use the Wait() methods, so you shouldn't change this to wait_for_element_displayed